src/GeolocatorService.Reactive/GeolocatorServiceExtensions.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System;
+using System.Linq;
+using System.Reactive.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Windows.Devices.Geolocation;
+
+namespace GeolocatorService
+{
+	public static class GeolocatorServiceExtensions
+	{
+		/// <summary>
+		/// Observes the user's location.
+		/// </summary>
+		public static IObservable<Geocoordinate> ObserveLocation(IGeolocatorService service)
+		{
+			return Observable.FromEvent<LocationChangedEventHandler, LocationChangedEventArgs>(
+				h => service.LocationChanged += h,
+				h => service.LocationChanged -= h
+			)
+			.Select(p => p.Location);
+		}
+
+		/// <summary>
+		/// Gets and observes the user's location. Asks the user for the location permission if this was not already done.
+		/// </summary>
+		public static async Task<IObservable<Geocoordinate>> GetAndObserveLocation(IGeolocatorService service, CancellationToken ct)
+		{
+			await service.RequestPermission(ct);
+			var currentLocation = await service.GetLocation(ct);
```

Comment:
if getLocation throws, the whole observable will end on error. Is it better to use the same one for this?